### PR TITLE
[pigeon] feat(pigeon): expose SwiftOptions

### PR DESCRIPTION
**Pull request by Lyokone**
Exposing SwiftOptions which are missing for pigeon.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [X] I signed the [CLA].
- [X] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [X] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [X] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by Lyokone** [comment]
Hello, since it's just an export, I'm not sure how to test that? 🤔

**Comment by stuartmorgan** [comment]
Presumably you are adding this for a reason; what were you trying to do that didn't work without that PR? That's what you should test.

**Comment by ailtonvivaz** [comment]
I don't know either how test it.
But maybe updating the `pigeons/message.dart` and adding new tests in `test/pigeon_test.dart` could validate?

**Comment by gaaclarke** [comment]
I think the easiest thing would be to have a dart unit test for this that imports the `package:pigeon/pigeon.dart` and references swiftoptions such that if you removed the line you added the test would no longer compile.

**Comment by stuartmorgan** [comment]
@Lyokone Are you planning on adding a test per the discussion above?

**Comment by Lyokone** [comment]
Hello, sorry I forgot about this. I'll do it today

File: packages/pigeon/CHANGELOG.md [hunk 1]
```diff
@@ -1,285 +1,289 @@
+## 3.2.10
+
+- Expose SwiftOptions
+
 ## 3.2.9
 
-* Updates text theme parameters to avoid deprecation issues.
+- Updates text theme parameters to avoid deprecation issues.
```
Comment (stuartmorgan) [inline packages/pigeon/CHANGELOG.md:7]: Please don't make changes that violate our style guide. We use `*`.

File: packages/pigeon/test/pigeon_test.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import 'package:pigeon/pigeon.dart';
```
Comment (stuartmorgan) [inline packages/pigeon/test/pigeon_test.dart:1]: This file needs a license header, as indicated by the failing CI step.

File: packages/pigeon/test/pigeon_test.dart [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import 'package:test/test.dart';
+
+void main() {
+  test('Should be able to import JavaOptions', () async {
+    const JavaOptions javaOptions = JavaOptions();
+    expect(javaOptions, isNotNull);
+  });
+  test('Should be able to import ObjcOptions', () async {
```
Comment (stuartmorgan) [inline packages/pigeon/test/pigeon_test.dart:9]: Please be consistent about adding a blank line between tests.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/pigeon/CHANGELOG.md (hunk 1 above)
Comment (Lyokone) [reply packages/pigeon/CHANGELOG.md]: Oops, autoformatting of VSCode a bit aggressive here; fixed

File: packages/pigeon/test/pigeon_test.dart (hunk 2 above)
Comment (Lyokone) [reply packages/pigeon/test/pigeon_test.dart:5]: Added the same licence as other files

File: packages/pigeon/test/pigeon_test.dart (hunk 3 above)
Comment (Lyokone) [reply packages/pigeon/test/pigeon_test.dart:14]: Added a blank space

**Review by tarrinneal — COMMENTED** [review]
If you can handle the conflicts (update this to the most recent version), and that last test passes, I'll get this merged for you.

**Comment by Lyokone** [comment]
@tarrinneal It's fixed :)

**Review by tarrinneal — APPROVED** [review]

**Review by stuartmorgan — APPROVED** [review]
I fixed the changelog to follow the changelog style guide (linked from the PR checklist). LGTM with that change.

**Pull request merged**